tools/klee/main.cpp

```diff
@@ -544,6 +527,36 @@ void KleeHandler::processTestCase(const ExecutionState &state,
           *f << PrettyExpressionBuilder::construct(*it);
       }
       delete f;
+
+      // Output the symbolic error
+      std::string errors = state.symbolicError->getOutputString();
+      if (!errors.empty()) {
+        llvm::raw_ostream *f = openTestFile("precision_error", id);
+        *f << errors;
+        delete f;
+      }
+
+      double pathProbability = state.symbolicError->getPathProbability();
+      int branchCount = state.symbolicError->getBranchCount();
+      if (pathProbability > 0) {
+        llvm::raw_ostream *probFile = openTestFile("prob", id);
+        *probFile << "(pathLength, pathProbability), " << id << "\n"
+                  << branchCount << ", " << pathProbability;
+        delete probFile;
+      }
+
+      llvm::raw_ostream *expression_file = openTestFile("expressions", id);
+      std::map<std::string, ref<Expr> > expressions =
```

Comment:
@Himeshi Here the type can be 
```
std::map<std::string, ref<Expr> > &
```
(with ampersand). Similar to the discussion above, this way is more efficient since we avoid copying.